### PR TITLE
br: wait more time to wait spitting the region

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/42001

Problem Summary:
The operation of scanning regions from PD exceeds the retry time. The old backoff is 128s， the new one is 5min

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Fix the issue, expand the backoff to scan regions from PD
```

### CODE REVIEW

File: br/pkg/restore/split/split.go [hunk 1]
```diff
@@ -192,9 +192,10 @@ func newScanRegionBackoffer() utils.Backoffer {
 // NextBackoff returns a duration to wait before retrying again
 func (b *scanRegionBackoffer) NextBackoff(err error) time.Duration {
 	if berrors.ErrPDBatchScanRegion.Equal(err) {
-		// 1s * 60 could be enough for splitting remain regions in the hole.
+		// it needs more time to wait splitting the regions that contains data in PITR.
+		// 2s * 150
 		b.attempt--
-		return time.Second
+		return time.Second * 2
```
Comment (hawkingrei) [inline br/pkg/restore/split/split.go:198]: Should we use exponential backoff？

File: br/pkg/restore/split/split.go [hunk 2]
```diff
@@ -174,27 +175,45 @@ func ScanRegionsWithRetry(
...
 
 // NextBackoff returns a duration to wait before retrying again
 func (b *scanRegionBackoffer) NextBackoff(err error) time.Duration {
 	if berrors.ErrPDBatchScanRegion.Equal(err) {
-		// 1s * 60 could be enough for splitting remain regions in the hole.
+		// it needs more time to wait splitting the regions that contains data in PITR.
+		// 2s * 150
 		b.attempt--
-		return time.Second
+		b.delayTime *= 2
```
Comment (YuJuncen) [inline br/pkg/restore/split/split.go:204]: You can embed `utils.RetryState` if you need to implement the Exponential Backoff. Just like 
```go
if berrors.ErrPDBatchScanRegion.Equal(err) {
  delayTime := state.ExponentialBackoff()
  failpoint.Inject(...)
  return delayTime
}
```

**Review by YuJuncen — COMMENTED** [review]
rest lgtm

**Review by Leavrth — APPROVED** [review]

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**